File: backend/app/schemas/application_record.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class ApplicationRecordCreate(BaseModel):
    resume_id: int
    job_posting_id: int
    current_stage: str = "saved"
    apply_channel: str | None = None
    applied_at: datetime | None = None
    next_action: str | None = None
    next_action_at: datetime | None = None
    notes: str | None = None


class ApplicationRecordUpdate(BaseModel):
    current_stage: str | None = None
    apply_channel: str | None = None
    applied_at: datetime | None = None
    next_action: str | None = None
    next_action_at: datetime | None = None
    notes: str | None = None


class ApplicationRecordListItem(BaseModel):
    id: int
    resume_id: int
    job_posting_id: int
    current_stage: str
    next_action: str | None
    next_action_at: datetime | None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class ApplicationRecordRead(ApplicationRecordListItem):
    apply_channel: str | None
    applied_at: datetime | None
    notes: str | None

File: frontend/src/types/application_record.ts
import type { ISODateString } from "./common";

export interface ApplicationRecordListItem {
  id: number;
  resume_id: number;
  job_posting_id: number;
  current_stage: string;
  next_action: string | null;
  next_action_at: ISODateString | null;
  created_at: ISODateString;
  updated_at: ISODateString;
}

export interface ApplicationRecord extends ApplicationRecordListItem {
  apply_channel: string | null;
  applied_at: ISODateString | null;
  notes: string | null;
}

export interface ApplicationRecordCreate {
  resume_id: number;
  job_posting_id: number;
  current_stage?: string;
  apply_channel?: string | null;
  applied_at?: ISODateString | null;
  next_action?: string | null;
  next_action_at?: ISODateString | null;
  notes?: string | null;
}

export type ApplicationRecordUpdate = Partial<ApplicationRecordCreate>;
